internal/overlord/servstate/handlers.go

```diff
@@ -117,25 +117,25 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 
 	// Wait for a small amount of time, and if the service hasn't exited,
 	// consider it a success.
-	timeout := time.After(okayWait + 100*time.Millisecond)
-	for {
-		select {
-		case err := <-service.started:
-			if err != nil {
-				m.removeService(config.Name)
-				return fmt.Errorf("cannot start service: %w", err)
-			}
-			// Started successfully (ran for small amount of time without exiting).
-			return nil
-		case <-tomb.Dying():
-			// Start operation cancelled (ignore for simplicity).
-			logger.Noticef("Ignoring cancellation of start for service %q", config.Name)
-		case <-timeout:
-			// Should never happen, because okayWaitElapsed and exited both send to the
-			// "started" channel, but don't block in case we got something wrong.
+	select {
+	case err := <-service.started:
+		if err != nil {
 			m.removeService(config.Name)
-			return fmt.Errorf("internal error: timed out waiting for start")
+			return fmt.Errorf("cannot start service: %w", err)
+		}
+		// Started successfully (ran for small amount of time without exiting).
+		return nil
+	case <-tomb.Dying():
+		// User tried to abort the start, sending SIGKILL to process is about
+		// the best we can do.
+		m.removeService(config.Name)
+		m.servicesLock.Lock()
+		defer m.servicesLock.Unlock()
+		err := syscall.Kill(-service.cmd.Process.Pid, syscall.SIGKILL)
+		if err != nil {
+			return fmt.Errorf("start aborted, but cannot send SIGKILL to process: %v", err)
 		}
+		return fmt.Errorf("start aborted, sent SIGKILL to process")
```

Comment:
Do we have context for which service is dying in these messages above, or do we need to add some sort of `service %q`?

Reply:
Yes. For example, if you press Ctrl-C in the 1s okay-wait period while running the `pebble start` CLI command, it calls "abort" on the change, and the error is wrapped with the service name. You get `Start service "test" (start aborted, sent SIGKILL to process)`.